travis/travis_weblate.py

```diff
@@ -1,166 +1,216 @@
+                       os.environ.get("LANG_ALLOWED", False) else [])
+        self._odoo_full = os.environ.get("ODOO_REPO", "odoo/odoo")
+        self._server_path = get_server_path(self._odoo_full,
+                                            (self._odoo_branch or
+                                             self._odoo_version),
+                                            self._travis_home)
+        self._addons_path = get_addons_path(self._travis_home,
+                                            self._travis_build_dir,
+                                            self._server_path)
+        self._addons_list = get_addons_to_check(self._travis_build_dir,
+                                                self._odoo_include,
+                                                self._odoo_exclude)
+        self._all_depends = (
+            get_depends(self._addons_path, self._addons_list) +
+            self._addons_list)
+        self._main_modules = set(os.listdir(self._travis_build_dir))
+        self._main_depends = self._main_modules & set(self._all_depends)
+        self._addons_list = list(self._main_depends)
+        self._connection_context = context_mapping.get(
+            self._odoo_version, Odoo10Context)
+        self._apply_patch_odoo()
+
+    def _check(self):
+        self.wl_api._check()
+        self.gh_api._check()
+
+    def _apply_patch_odoo(self):
+        """This patch is necessary because the weblate does not check which
+        word and the translated are the same to use it in its percentage of
+        translated"""
+        paths = [os.path.join('openerp', 'tools', 'translate.py'),
+                 os.path.join('odoo', 'tools', 'translate.py')]
+        for path in paths:
+            s_file = os.path.join(self._server_path, path)
+            if not os.path.isfile(s_file):
+                continue
+            cmd = ("sed -i -e \"s/row\\['translation'] = src/"
+                   "row['translation'] = ''/g\" %s" % s_file)
+            print cmd
+            subprocess.call(cmd, shell=True)
+
+    def _generate_odoo_po_files(self):
+        with self._connection_context(self._server_path, self._addons_path,
+                                      "openerp_test") as odoo_context:
+            for module in self._addons_list:
+                print("\n", yellow("Obtaining POT file for %s" % module))
+                i18n_folder = os.path.join(self._travis_build_dir, module,
+                                           'i18n')
+                if not os.path.isdir(i18n_folder):
+                    os.makedirs(i18n_folder)
+                # Put git add for letting known git which translations to update
+                po_files = glob.glob(os.path.join(i18n_folder, '*.po'))
+                for lang in self._langs:
+                    if os.path.isfile(os.path.join(i18n_folder, lang + '.po')):
+                        continue
+                    with open(os.path.join(i18n_folder, lang + '.po'), 'wb') \
```

Comment:
check if the module is really installed because this could delete the content of the file